Introduce find_mutation and mutation_stats class

stevenhwu opened the pull request:
Replace #104 
find_mutation: used in call.cc, call peeling operation, and pass workspace to mutation_stats. Some unit tests.

mutation_stats: calculate and store all stats. Store stats to vcf. Some unit tests
TODO(SW): need an interface between mutation_stats and outfile(various format)
TODO(SW): Later on will have switches allow users only output a subsets of stats. Still working on moving some counts related stats from call.cc to here.

Refactoing peeling.cc. Issue #80. Fixed merge issue on #83 

test/src/fixture/\* - Collection of fixtures can be used in multiple boost_tests.

reedacartwright committed:
Improved Support for Regions in Bam Files

 - Create our own region specification format.
   - chr1:1-100, chr1:1-, chr1:-100, chr1:100, chr1:1+100
 - Use our format instead of htslib's
 - Support reading bam files across multiple regions
 - Add a dng/utility.h and dng/io/utility.h files
 - Allow reading regions from file

Fixes #42

added location_t typedef

added mask to dng::utility::make_location for position

began playing with regions

Initial parsing of region works

update regions parser and tests

Updated region parsing

begin work to add multiple regions to hts::bam::File

more work on regions

handle converting raw, parsed regions into bam regions

multiple regions now work with call

added thousands separator support

update character space

added io utilities and the ability to read regions from a file if the argument begins with @

stevenhwu committed:
Introduce find_mutation and mutation_stats class

find_mutation: used in call.cc, call peeling operation, and pass workspace to mutation_stats. Some unit tests.

mutation_stats: calculate and store all stats. Store stats to vcf. Some unit tests
TODO(SW): need an interface between mutation_stats and outfile(various format)

stevenhwu committed:
Introduce find_mutation and mutation_stats class

find_mutation: used in call.cc, call peeling operation, and pass workspace to mutation_stats. Some unit tests.

mutation_stats: calculate and store all stats. Store stats to vcf. Some unit tests
TODO(SW): need an interface between mutation_stats and outfile(various format)

dng-jenkins commented:
Can one of the admins verify this patch?

reedacartwright commented:
Jenkins, ok to test
On Mar 9, 2016 1:34 AM, "Steven Wu" <notifications@github.com> wrote:

> Replace #104 <https://github.com/denovogear/denovogear/pull/104>
> find_mutation: used in call.cc, call peeling operation, and pass workspace
> to mutation_stats. Some unit tests.
>
> mutation_stats: calculate and store all stats. Store stats to vcf. Some
> unit tests
> TODO(SW): need an interface between mutation_stats and outfile(various
> format)
> TODO(SW): Later on will have switches allow users only output a subsets of
> stats. Still working on moving some counts related stats from call.cc to
> here.
>
> Refactoing peeling.cc. Issue #80
> <https://github.com/denovogear/denovogear/issues/80>. Fixed merge issue
> on #83 <https://github.com/denovogear/denovogear/pull/83>
>
> test/src/fixture/* - Collection of fixtures can be used in multiple
> boost_tests.
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/denovogear/denovogear/pull/120
> Commit Summary
>
>    - Introduce find_mutation and mutation_stats class
>
> File Changes
>
>    - *M* src/CMakeLists.txt
>    <https://github.com/denovogear/denovogear/pull/120/files#diff-0> (1)
>    - *M* src/dng-call.cc
>    <https://github.com/denovogear/denovogear/pull/120/files#diff-1> (1)
>    - *A* src/include/dng/find_mutation.h
>    <https://github.com/denovogear/denovogear/pull/120/files#diff-2> (126)
>    - *M* src/include/dng/graph.h
>    <https://github.com/denovogear/denovogear/pull/120/files#diff-3> (10)
>    - *A* src/include/dng/mutation_stats.h
>    <https://github.com/denovogear/denovogear/pull/120/files#diff-4> (148)
>    - *M* src/include/dng/pedigree.h
>    <https://github.com/denovogear/denovogear/pull/120/files#diff-5> (16)
>    - *M* src/include/dng/peeling.h
>    <https://github.com/denovogear/denovogear/pull/120/files#diff-6> (195)
>    - *M* src/lib/call.cc
>    <https://github.com/denovogear/denovogear/pull/120/files#diff-7> (460)
>    - *A* src/lib/find_mutation.cc
>    <https://github.com/denovogear/denovogear/pull/120/files#diff-8> (367)
>    - *A* src/lib/mutation_stats.cc
>    <https://github.com/denovogear/denovogear/pull/120/files#diff-9> (313)
>    - *M* src/lib/peeling.cc
>    <https://github.com/denovogear/denovogear/pull/120/files#diff-10>
>    (219)
>    - *A* src/utils/boost_utils.h
>    <https://github.com/denovogear/denovogear/pull/120/files#diff-11> (39)
>    - *A* src/utils/vcf_utils.h
>    <https://github.com/denovogear/denovogear/pull/120/files#diff-12>
>    (174)
>    - *M* test/CMakeLists.txt
>    <https://github.com/denovogear/denovogear/pull/120/files#diff-13>
>    (118)
>    - *D* test/boost_test_helper.h
>    <https://github.com/denovogear/denovogear/pull/120/files#diff-14> (65)
>    - *A* test/src/boost_test_helper.h
>    <https://github.com/denovogear/denovogear/pull/120/files#diff-15>
>    (110)
>    - *A* test/src/fixture/fixture_random_family.h
>    <https://github.com/denovogear/denovogear/pull/120/files#diff-16>
>    (163)
>    - *A* test/src/fixture/fixture_read_trio_from_file.h
>    <https://github.com/denovogear/denovogear/pull/120/files#diff-17>
>    (115)
>    - *A* test/src/fixture/fixture_trio_workspace.h
>    <https://github.com/denovogear/denovogear/pull/120/files#diff-18>
>    (108)
>    - *A* test/src/lib/test2_find_mutation.cc
>    <https://github.com/denovogear/denovogear/pull/120/files#diff-19>
>    (336)
>    - *A* test/src/lib/test2_mutation_stats.cc
>    <https://github.com/denovogear/denovogear/pull/120/files#diff-20>
>    (237)
>    - *R* test/src/lib/test2_peeling.cc
>    <https://github.com/denovogear/denovogear/pull/120/files#diff-21>
>    (303)
>
> Patch Links:
>
>    - https://github.com/denovogear/denovogear/pull/120.patch
>    - https://github.com/denovogear/denovogear/pull/120.diff
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/denovogear/denovogear/pull/120>.
>

stevenhwu committed:
Fix jenkins error

Changed from "free" to 'std::free' at src/include/dng/hts/hts.h:56:95:

stevenhwu committed:
Fix jenkins error

Changed from "free" to 'std::free' at src/include/dng/hts/hts.h:56:95:

stevenhwu committed:
Add missing header #include <cstdlib>

Need to test this on Jenkins.
Working locally with/without std::, with/without #include <cstdlib>

stevenhwu committed:
Add missing header #include <cstdlib>

Need to test this on Jenkins.
Working locally with/without std::, with/without #include <cstdlib>

stevenhwu commented:
Why this error appear now but not before? Were some headers got included implicitly?

Address Jenkins' error
```c++
In file included from /usr/local/jenkins/jobs/DeNovoGear-PR/workspace/src/include/dng/hts/bcf.h:23:
/usr/local/jenkins/jobs/DeNovoGear-PR/workspace/src/include/dng/hts/hts.h:56:95: error: use of undeclared identifier 'free'
        std::unique_ptr<char[], void(*)(void *)> s{hts_format_description(&handle()->format), free};
                                                                                              ^
```

Fixed with `#include <cstdlib>` in `dng/hts/hts.h`

stevenhwu commented:
Just want to double check this, does Jenkin ran `make test` or see if code in the test folder compiles without error?
I can find `[build] $ /usr/bin/make` but not `make other_targets` in Jenkins' log.

reedacartwright commented:
Currently, our Jenkins instance runs on a FreeBSD 10 machine with Clang 3.4 and libc++.  Something that works on Linux development boxes can fail on the Jenkins machine.

Additionally, our Jenkins instance just checks for successful compiling and not successful tests. As soon as #35 is solved, we will do more.

stevenhwu commented:
but `src/include/dng/hts/hts.h` was last modified on June 2015 and pass all previous tests/compile. Are there any ways to capture more issues like this?

reedacartwright commented:
The problem was masked earlier due to `cstdlib` being included before `hts.h`.  But the new `mutation_stats.cc` file exposed it.

As far as fixing these, we when doing unit testing on headers, we should make the header the first one included.

reedacartwright committed:
support boost 1.55

reedacartwright committed:
typo of previous commit

reedacartwright committed:
Merge pull request #121 from reedacartwright/feature/42-newregions-squashed

Improved Support for Regions in Bam Files

reedacartwright committed:
fix runtime errors with boost 1.55

stevenhwu committed:
rebased on develop 701614ab

stevenhwu committed:
Fixed merge bug

reedacartwright commented on src/include/dng/mutation_stats.h:
```diff
@@ -0,0 +1,148 @@
+
+    enum class OutputLevel {
+        Basic, Complete, Singcle
+    };
```
Misspelling

dng-jenkins commented:
Can one of the admins verify this patch?
